File: game.py
pwin=input();
ploose=1-pwin;
q=input();
while q:
    n=input();
    i=2;
    pw=pwin;
    pl=ploose;
    while i<=n:
        pw=pw*ploose+pl*pwin;
        pl=1-pw;    
        i=i+1;
    print("%.4f" % pw)
    q=q-1;
    0570101002246899

File: game.c
#include<stdio.h>

float pwin,ploose;
float win(int q)
{
    float winprev;
    if(q==1)
    {
        return pwin;
    }
    else
    {
        winprev=win(q-1);
        return winprev*ploose+(1-winprev)*pwin;
    }

}
int main()
{
    float n,q;
    float x;
    scanf("%f%f",&pwin,&n);
    ploose=1-pwin;
    while(n)
    {
        scanf("%f",&q);
        x=win(q);    
        printf("\n%.4f",x);
        n--;
    }
    return 0;

}
